tests/unit/test_wait_for_idle.py

```diff
@@ -0,0 +1,540 @@
+# Copyright 2024 Canonical Ltd.
```

Comment:
I like the new unit tests, just a note about testing the case where `apps` is a `str` since old implementation raises an error here that I don't believe the new implementation does (instead silently timing out?)

Reply:
done